common/kvstore/batch.go

```diff
@@ -0,0 +1,32 @@
+package kvstore
+
+import "time"
+
+// Batch is a collection of key / value pairs that will be written atomically to a database.
+// Although it is thread safe to modify different batches in parallel or to modify a batch while
+// the store is being modified, it is not thread safe to concurrently modify the same batch.
+type Batch[K any] interface {
+	// Put stores the given key / value pair in the batch, overwriting any existing value for that key.
+	// If nil is passed as the value, a byte slice of length 0 will be stored.
+	Put(key K, value []byte)
+	// Delete removes the key from the batch.
+	Delete(key K)
+	// Apply atomically writes all the key / value pairs in the batch to the database.
+	Apply() error
+	// Size returns the number of operations in the batch.
+	Size() uint32
+}
+
+// TTLBatch is a collection of key / value pairs that will be written atomically to a database with
+// time-to-live (TTL) or expiration times. Although it is thread safe to modify different batches in
+// parallel or to modify a batch while the store is being modified, it is not thread safe to concurrently
+// modify the same batch.
+type TTLBatch[K any] interface {
```

Comment:
It looks this is not needed?
- if it's a `Table`, there is `TableBatch`, operating on the raw keys
- if it's a `TableStore`, there is `TableStoreBatch`, operating on the namespaced keys

And the ttl related interfaces are always there.

Reply:
The reason why we need a batch with TTLs and a batch without TTLs is because `Store` supports batches but has no concept of TTLs

In the previous PR, you requested that I have a `Batch[K any]` interface that is inherited by the subtypes `StoreBatch[[]byte]` and `TableStoreBatch[TableKey]`. The generic interface `TTLBatch[K any] is a continuation of that pattern.

I agree that these interfaces are currently kind of clunky. But I'm not sure how to make them better while still respecting some of your prior requests. We should discuss strategies for improving this.

Reply:
The batch APIs have now been simplified so that they don't have generics. Are there additional simplifications you'd like to see?

Reply:
Note that we still need two batch interfaces if we want to support batches that can set different TTLs for operations within the same batch. The core reason is that the basic `Store` API does not have a concept of a TTL, meaning we need a batch interface that doesn't have TTLs.